api4/post.go

```diff
@@ -416,7 +421,58 @@ func getPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// getPostNew will replace the getPost
+func getPostNew(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequirePostId()
+	if c.Err != nil {
+		return
+	}
+
+	includeDeleted, _ := strconv.ParseBool(r.URL.Query().Get("include_deleted"))
+	if includeDeleted && !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSystem) {
+		c.SetPermissionError(model.PermissionManageSystem)
+		return
+	}
+
+	postWrapper, err := c.App.GetPostIfAuthorizedNew(c.Params.PostId, c.AppContext.Session(), includeDeleted)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if postWrapper.IsInaccessible {
+		if err := json.NewEncoder(w).Encode(postWrapper); err != nil {
+			mlog.Warn("Error while writing response", mlog.Err(err))
+		}
+		return
+	}
+
+	post := postWrapper.Post
+	post = c.App.PreparePostForClientWithEmbedsAndImages(post, false, false)
+	post, err = c.App.SanitizePostMetadataForUser(post, c.AppContext.Session().UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+	postWrapper.Post = post
+
+	if c.HandleEtag(post.Etag(), "Get Post", w, r) {
+		return
+	}
+
+	w.Header().Set(model.HeaderEtagServer, post.Etag())
+	post.StripActionIntegrations()
```

Comment:
Actually, the old code already does it in this abstraction - `post.EncodeJSON(w)`. But since I'm using `PostWrapper` now, I need to write the same functionality explicitly here.